File: 04 Kontrollstrukturen/Go/Aufgabe8.go
package main

import "fmt"

var (
	zahl int
	fakultaet int = 1
	i int
)

func main() {
	fmt.Print("Bitte gib deine Zahl ein: ")
	fmt.Scan(&zahl)

	if zahl < 0 {
		fmt.Println("Die Fakultät von negativen Zahlen ist nicht definiert!")
	} else if zahl == 0 {
		fmt.Println("Eigentlich ist 0! = 1, aber laut Aufgabenstellung soll es")
		fmt.Println("eine Fehlermeldung geben. Also: Böse böse!")
	} else {
		for i = 0; i < zahl; i++ {
			fakultaet *= i + 1
		}

		fmt.Println("Die Fakultät von", zahl, "ist", fakultaet, ".")
	}
}

File: 04 Kontrollstrukturen/Pascal/Aufgabe8.pas
Program Aufgabe8;

{$codepage utf8}

Var Zahl: Integer;
    Fakultaet: Integer = 1;
    I: Integer;

Begin
    Write('Bitte gib deine Zahl ein: ');
    Read(Zahl);

    If Zahl < 0 Then
        WriteLn('Die Fakultät von negativen Zahlen ist nicht definiert!')
    Else If Zahl = 0 Then Begin
        WriteLn('Eigentlich ist 0! = 1, aber laut Aufgabenstellung soll es');
        WriteLn('eine Fehlermeldung geben. Also: Böse böse!');
    End Else Begin
        For I := 1 To Zahl Do
            Fakultaet *= I;
        
        WriteLn('Die Fakultät von ', Zahl, ' ist ', Fakultaet, '.');
    End
End.